help/md/mcmc.md

```diff
@@ -11,28 +11,41 @@ An MCMC analysis is initiated using the `mcmc.run()` method.
 The `StoppingRule[]` argument provides a mechanism to automatically terminate an MCMC run once a set of rules are met: perhaps once the run has attained convergence, or after a certain amount of time has passed.  The run will be terminated once *all* convergence rules ([`srGelmanRubin()`], [`srGeweke()`], [`srMinESS()`], [`srStationarity()`]) have been fulfilled; or once *any* threshold rules ([`srMaxTime()`], [`srMaxIteration()`]) are met.
 The parameters `checkpointFile` and `checkpointInterval` generate snapshots of the current state of the MCMC run from which the run can be continued if interrupted using the `mcmc.initializeFromCheckpoint()` method.
 
+The `mcmc.initializeFromCheckpoint()` method allows an analysis to be continued from a checkpoint file.
+New generations will be appended to existing monitor files.
+
 ## authors
 Sebastian Hoehna
 ## see_also
 mcmcmc
 ## example
-	# Create a simple model (unclamped)
-	a ~ exponential(1)
-	mymodel = model(a)
-	
-	# Create a move vector and a monitor vector
-	moves[1] = mvScale(a, lambda=1.0, weight=1.0)
-	monitors[1] = mnFile(a,"output/out.log")
-	
-	# Create an mcmc object
-	mymcmcObject = mcmc( mymodel, monitors, moves)
-	
-	# Run a short analysis
-	mymcmcObject.burnin( generations = 400, tuningInterval = 100)
-	mymcmcObject.run( generations = 400)
-	
-	# print the summary of the operators (now tuned)
-	mymcmcObject.operatorSummary()
+    # Create a simple model (unclamped)
+    a ~ dnExponential(1)
+    mymodel = model(a)
+    
+    # Create a move vector and a monitor vector
+    moves[1] = mvScale( a, lambda = 1.0, weight = 1.0 )
+    monitors[1] = mnFile( a, filename = "output/out.log" )
+    
+    # Create an mcmc object
+    mymcmcObject = mcmc( mymodel, monitors, moves )
+    
+    # Run a short analysis
+    mymcmcObject.burnin( generations = 400, tuningInterval = 100 )
+    mymcmcObject.run( generations = 400, checkpointFile = "output/out.ckp", checkpointInterval = 100 )
+    
+    # print the summary of the operators (now tuned)
+    mymcmcObject.operatorSummary()
+
+    # Resume analysis from the checkpoint file
+    mymcmcObject.initializeFromCheckpoint( "output/out.ckp" )
+
+    # Conduct an additional 400 generations
+    mymcmcObject.run( generations = 400 )
+
+    # Stopping rules are defined on the total number of generations
+    # This command will have no effect, as 400 generations have already been performed.
+    mymcmcObject.run( generations = 400, rules = [ srMaxIteration(400) ] )
```

Comment:
Ah, so there _is_ a difference between `generations` and `srMaxIteration` after all! Congratulations on uncovering this :) However, maybe the final line is a bit too complicated, and

```
mymcmcObject.run( rules = [ srMaxIteration(400) ] )
```

would suffice to illustrate the point?

Reply:
I've simplified the example, and rewritten the text to clarify the interaction between the parameters.